hs_geographic_feature_resource/receivers.py

```diff
@@ -57,13 +60,22 @@ def check_fn_for_shp(files):
                 fileName, fileExtension = os.path.splitext(fullName.lower())
                 if ".shp" == fileExtension:
                     shp_filename = fileName
-                    shp = True
+                    if not shp:
+                        shp = True
+                    else:
+                        return False
```

Comment:
@hyi 
Sorry... I said it wrong above. 
The variable 'shp' is defined inside this function, so it is a local variable, not global. As it is defined outside the for loop, each cycle accesses the same variable 'shp'. So we use it to detect duplicate any '.shp' file in loop.